pages/ingredient/[id]/edit.js

```diff
@@ -0,0 +1,100 @@
+import { useRouter } from "next/router";
+import { useState, useEffect } from "react";
+import {
+  Container,
+  StyledContent,
+  BackLink,
+  EditForm,
+  Input,
+  SaveButton,
+  CancelButton,
+} from "@/_styles.js";
+
+const EditIngredient = ({ ingredients, updateIngredient }) => {
+  const router = useRouter();
+  const { id } = router.query;
+  const ingredient = ingredients.find((ingredient) => ingredient._id === id);
+
+  const [formData, setFormData] = useState({
+    name: "",
+    imgUrl: "",
+    flavor: "",
+  });
+
+  useEffect(() => {
+    if (ingredient) {
+      setFormData({
+        name: ingredient.name,
+        imgUrl: ingredient.imgUrl,
+        flavor: ingredient.flavor,
+      });
+    }
+  }, [ingredient]);
```

Comment:
You don't need a useEffect here, this will work just as fine:

```suggestion
const [formData, setFormData] = useState({
    name: ingredient.name || "",
    imgUrl: ingredient.imgUrl || "",
    flavor: ingredient.flavor || "",
  });
```